src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,36 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+	public final String value;
+	private boolean isPrivate;
+
+	public Contact(String contact, boolean isPrivate, String errorMessage, String regex) throws IllegalValueException {
```

Comment:
Interesting design by capturing the validation regex and messages in constructor.